Add sustainability modules to mapping documentation

duncandewhurst opened the pull request:
**Related issues**

* Closes #395

**Description**

The mapping is stored in `sustainability.yaml` and there are two new `manage.py` commands:

* `update-sustainability-docs` generates the Markdown content in `docs/cost/ids/sustainability.md` using `sustainability.yaml`.
* `lint` formats and validates the data in `sustainability.yaml` and reports missing data

Presently, `lint` reports many additional fields and some missing data. This is expected because the schema is not yet updated and because the mapping for some elements is still under discussion.

However, I suggest that we merge this PR as-is and then, as part of each schema or codelist change PR, we can update the relevant mappings and examples in `sustainability.yaml`.

Before releasing, OC4IDS 0.9.4 we can do a final copy edit of the mappings and examples for consistency and check that `lint` reports no additional fields or missing data. I'll create a separate issue about that.

The spell-check is failing on the "TNE" unit code referenced in the mapping. @jpmckinney is there somewhere that needs to be added to a dictionary?

**Merge checklist**

<!-- Complete the checklist before requesting a review. -->

- [x] [Log your changes](https://ocds-standard-development-handbook.readthedocs.io/en/latest/standard/contributing.html#logging-changes)

If there are changes to `project-schema.json` or `project-package-schema.json`:

- [ ] Update the examples:
  - [ ] `docs/examples/example.json`
  - [ ] `docs/examples/blank.json`
- [ ] Run `./manage.py pre-commit` to update `docs/_static/i8n.csv`

If you added a new definition to the schema, run `./manage.py pre-commit`.

If you added a new codelist:

  - [ ] Add an entry to `docs/reference/codelists.md`

duncandewhurst committed:
docs/cost: Restructure mapping documentation, add sustainability page

duncandewhurst committed:
docs/cost/ids/sustainability.md: Add PDF download instructions

duncandewhurst committed:
Add mapping/sustainability.yaml

duncandewhurst committed:
mapping/sustainability.yaml: Fix syntax error

duncandewhurst committed:
mapping/sustainability.yaml: Lint Markdown

duncandewhurst committed:
mapping/sustainability.yaml: Lint examples

duncandewhurst committed:
docs/cost/ids/sustainability.md: Add mapping documentation

duncandewhurst committed:
manage.py: Add lint and update-sustainability-docs commands

duncandewhurst committed:
manage.py: Sort imports

duncandewhurst committed:
Update changelog

duncandewhurst committed:
manage.py: Add comments

duncandewhurst committed:
docs/cost/ids/sustainability.md: Correct typos

duncandewhurst committed:
mapping/sustainability.yaml: Remove placeholder text

duncandewhurst committed:
mapping/sustainability.yaml: Remove placeholder text

jpmckinney commented:
For spellcheck, change `-L zar` to `-L tne,zar` in `spellcheck.yml`.

jpmckinney commented on docs/cost/ids/core.md:
```diff
@@ -0,0 +1,267 @@
+
+## Project level
+
+## Identification
```
The heading levels below level 2 don't seem correct.

duncandewhurst replied:
Fixed in https://github.com/open-contracting/infrastructure/pull/422/commits/c6104c6125a01841151d4f10f2d875b438014aac

jpmckinney reviewed (commented):
Looks like a straight split of the content (i.e. without additional changes), plus the addition of the sustainability content (which will continue to change).

If that's the case, then my only comment is with heading levels having changed.

duncandewhurst committed:
.github/workflows/spellcheck.yml: Ignore 'TNE'

duncandewhurst committed:
docs/cost/ids/core.md: Fix heading levels

duncandewhurst commented:
> Looks like a straight split of the content (i.e. without additional changes), plus the addition of the sustainability content (which will continue to change).

Yep, that's correct.

jpmckinney commented on docs/cost/ids/core.md:
````diff
@@ -0,0 +1,267 @@
+:file: ../../../build/current_lang/reactive-project-level-completion.csv
+```
+
+#### Implementation progress reports
````
This was originally one heading level lower than Completion

duncandewhurst replied:
Ah, good spot. Fixed in https://github.com/open-contracting/infrastructure/pull/422/commits/c71446fb5f86849bc7575b9811164ed53e27a159

jpmckinney commented on docs/cost/ocds.md:
```diff
@@ -0,0 +1,150 @@
+
+## Guidance
+
+## Command-line tool and reference implementation
```
This file also has most of the indentation incorrect.

duncandewhurst replied:
Oh dear. That'll teach me to use find and replace to update heading levels! Fixed in https://github.com/open-contracting/infrastructure/pull/422/commits/4df60d29b313b457a3b943f510c7901edda41806

jpmckinney replied:
FWIW I tend to replace `^##` with `#` (can also do `#(#+)` with `$1`) to reduce the heading level (syntax might differ across editors).

duncandewhurst replied:
Nice. Thanks!

duncandewhurst committed:
docs/cost/ocds.md: Fix heading levels, add introductory sentence

duncandewhurst committed:
docs/cost/ids/core.md: Fix heading levels, add introductory sentence

duncandewhurst committed:
docs/cost/ids/index.md: Add content

duncandewhurst committed:
docs/cost/index.md: Use sentence case, move some content to sub-pages

duncandewhurst commented:
I realised that `docs/cost/ids/index.md` was missing any content so I've added a description of its sub-pages. At the same time, I added introductory sentences to other pages under `docs/cost` and updated page titles to use sentence case.

jpmckinney commented on docs/cost/ids/core.md:
```diff
@@ -0,0 +1,271 @@
+}
+</style>
+
+This page documents a mapping from the 'core' CoST IDS elements to OC4IDS fields and codes. The organization of the mapping tables reflects the structure of the CoST IDS.
```
In case I close https://github.com/open-contracting/standard-maintenance-scripts/issues/10 (done for extensions, but not for standards or profiles), let's use double quotes for colloquial quotes.

```suggestion
This page documents a mapping from the "core" CoST IDS elements to OC4IDS fields and codes. The organization of the mapping tables reflects the structure of the CoST IDS.
```

jpmckinney commented on docs/cost/ids/core.md:
```diff
@@ -0,0 +1,271 @@
+
+This page documents a mapping from the 'core' CoST IDS elements to OC4IDS fields and codes. The organization of the mapping tables reflects the structure of the CoST IDS.
+
+The CoST IDS sets out a number of disclosure requirements under the heading of 'information for disclosure upon request', also known as 'reactive disclosure'. You can disclose these elements proactively using OC4IDS. Separate tables are provided for project-level and process-level reactive disclosures.
```
As previous comment, plus quotation marks go outside punctuation, even if that punctuation doesn't appear in the quoted material.

```suggestion
The CoST IDS sets out a number of disclosure requirements under the heading of "information for disclosure upon request," also known as "reactive disclosure." You can disclose these elements proactively using OC4IDS. Separate tables are provided for project-level and process-level reactive disclosures.
```

duncandewhurst replied:
I continue to expand my knowledge of the differences between American and British English!

jpmckinney commented on docs/cost/ids/core.md:
```diff
@@ -0,0 +1,271 @@
+
+This page documents a mapping from the 'core' CoST IDS elements to OC4IDS fields and codes. The organization of the mapping tables reflects the structure of the CoST IDS.
+
+The CoST IDS sets out a number of disclosure requirements under the heading of 'information for disclosure upon request', also known as 'reactive disclosure'. You can disclose these elements proactively using OC4IDS. Separate tables are provided for project-level and process-level reactive disclosures.
```
Should this paragraph also be present in `ocds.md`?

duncandewhurst replied:
Yes! Added in https://github.com/open-contracting/infrastructure/pull/422/commits/a198afb8f399543876b2b07ff0cd56481760c398

jpmckinney commented on docs/cost/ocds.md:
````diff
@@ -0,0 +1,152 @@
+:file: ../../build/current_lang/reactive-project-level-identification-preparation.csv
+```
+
+###### Completion
````
```suggestion
##### Completion
```

jpmckinney commented on docs/cost/ids/index.md:
```diff
@@ -0,0 +1,15 @@
+# CoST IDS to OC4IDS mapping
+
+This section contains a mapping from the CoST IDS to OC4IDS:
+
+* The [core standard mapping](core) describes a mapping of the proactive and reactive disclosure elements of the 'core' CoST IDS .
+* The [sustainability modules mapping](sustainability ) describes a mapping of the optional CoST IDS sustainability modules.
```
```suggestion
This section contains mappings from the CoST IDS to OC4IDS:

* The [core standard mapping](core) describes a mapping of the proactive and reactive disclosure elements of the "core" CoST IDS.
* The [sustainability modules mapping](sustainability) describes a mapping of the optional CoST IDS sustainability modules.
```

jpmckinney reviewed (commented):
Just a few small changes, then good to merge :)

duncandewhurst committed:
Apply suggestions from code review

Co-authored-by: James McKinney <26463+jpmckinney@users.noreply.github.com>

duncandewhurst committed:
docs/cost/ocds.md: Add note on reactive disclosures
